man/magellan.1.sc

```diff
@@ -79,11 +79,102 @@ The *magellan* command accepts
+4. Crawl Workflow:		crawl -> send -> crawl -> send -> *
+
+### Simple Workflow
+
+This is the simplest and most minimalistic way for using *magellan*. This method
+does not use the secrets store and no changes are made after an inventory
+collection.
+
+```
+magellan scan --subnet 172.16.0.0/24
+magellan collect -u $u -p $p | magellan send https://smd.example.com
+```
+
+### Complex Workflow
+
+This workflow is more complex and will include using a local cache database, 
+using the secret store, and allowing for editting the inventory collection before 
+sending it to our state manager.
+
+```
+# scan for assets on multiple networks
+magellan scan 10.0.0.101 \
+	--subnet 172.16.0.0/24 \
+	--subnet 172.21.0.0 \
+	--subnet-mask 255.255.255.0
+	--port 5000
+	--cache ./assets.db
+
+# show which hosts we found
+magellan list --cache ./assets.db
+
+# store secrets for host (saves to 'secrets.json' by default)
+magellan secrets store default $default_username:$default_password
+magellan secrets store $bmc_host1 $bmc_username1:$bmc_password1
+magellan secrets store $bmc_host2 $bmc_username2:$bmc_password2
+
+# perform collect using secrets store and save to YAML
+magellan collect --secrets-file secrets.json -o nodes.yaml -F yaml
+
+# make edits to inventory data
+vim nodes.yaml
+
+# read editted inventory and send data to host
+magellan send -d @nodes.yaml -F yaml https://smd.example.com
+```
+
+### Collect Workflow
+
+If we already have cache data, we can completely bypass doing a scan. Instead,
+we can repeatedly do a collect with little effort if we already have the secrets
+store set up and send the data to our remote host all in one step like in the
+"Simple Workflow".
+
+```
+# assume we already have a cache database and secrets store somewhere...
+magellan collect --secrets_file secrets.json | magellan send https://smd.example.com
+
+# alternatively, just update the file we send our output
+magellan collect --secrets-file secrets.json -o nodes.json
+```
+
+We can continuously do this whenever we want to update the state of our inventory.
+
+### Crawl Workflow
+
+This workflow is similar to the "Collect" workflow except we crawl a single BMC
+that we specify instead of multiple BMCs found from the scan. Like with the previous
+workflow, this can be done repeatedly to update the state of the inventory.
+
+```
+magellan crawl -i -u $u -p $p https://bmc.example.com -o node.json
+magellan send -d @node.json https://smd.example.com
+```
+
+Note that this only uses a single go routine instead of multiple like with *collect*.
+
+## References
+
+For more information about Redfish, visit https://https://redfish.dmtf.org/.
+For Redfish specifications and other documents, visit https://www.dmtf.org/standards/redfish.
 
 ## CONFIGURATION
 
-Nothing here yet.
+The *magellan* CLI configuration is handled by passing the *--config* in a single 
+YAML file. An example file can be found at in the root repository named
+*example.config.yaml*.
```

Comment:
At some point, it would be useful to have a section 5 man page on the Magellan config file detailing format and available options.

Reply:
Agreed, just putting it out there to be complete.